tests/src/Kernel/Entity/GetBundleByBundleTest.php

```diff
@@ -0,0 +1,557 @@
+   *   bundles that need to be set up in the test.
+   * @param array $expected_group_by_group_content
+   *   An array containing the expected results for the call to
+   *   getGroupBundleIdsByGroupContentBundle().
+   * @param array $expected_group_content_by_group
+   *   An array containing the expected results for the 4 calls to
+   *   getGroupContentBundleIdsByGroupBundle() that will be made in the test.
+   *
+   * @covers ::getGroupBundleIdsByGroupContentBundle
+   * @covers ::getGroupContentBundleIdsByGroupBundle
+   *
+   * @dataProvider getBundleIdsByBundleProvider
+   */
+  public function testGetBundleIdsByBundle(array $relationships, array $expected_group_by_group_content, array $expected_group_content_by_group) {
+    // Set up the relations as indicated in the test.
+    foreach ($relationships as $group_content_entity_type_id => $group_content_bundle_ids) {
+      foreach ($group_content_bundle_ids as $group_content_bundle_id => $group_audience_fields) {
+        switch ($group_content_entity_type_id) {
+          case 'node':
+            NodeType::create([
+              'name' => $this->randomString(),
+              'type' => $group_content_bundle_id,
+            ])->save();
+            break;
+          case 'block_content':
+            BlockContentType::create(['id' => $group_content_bundle_id])->save();
+            break;
+        }
+        foreach ($group_audience_fields as $group_audience_field_key => $group_audience_field_data) {
+          foreach ($group_audience_field_data as $group_entity_type_id => $group_bundle_ids) {
+            $settings = [
+              'field_name' => 'group_audience_' . $group_audience_field_key,
+              'field_storage_config' => [
+                'settings' => [
+                  'target_type' => $group_entity_type_id,
+                ],
+              ],
+            ];
+
+            if (!empty($group_bundle_ids)) {
+              $settings['field_config'] = [
+                'settings' => [
+                  'handler_settings' => [
+                    'target_bundles' => array_combine($group_bundle_ids, $group_bundle_ids),
+                  ],
+                ],
+              ];
+            }
+            Og::createField(OgGroupAudienceHelper::DEFAULT_FIELD, $group_content_entity_type_id, $group_content_bundle_id, $settings);
+          }
+        }
+      }
+    }
+
+    // Test ::getGroupBundleIdsByGroupContentBundle().
+    foreach ($expected_group_by_group_content as $group_content_entity_type_id => $group_content_bundle_ids) {
+      foreach ($group_content_bundle_ids as $group_content_bundle_id => $expected_result) {
+        $this->assertEquals($expected_result, $this->groupManager->getGroupBundleIdsByGroupContentBundle($group_content_entity_type_id, $group_content_bundle_id));
+      }
+    }
+
+    // Test ::getGroupContentBundleIdsByGroupBundle().
+    foreach (['node', 'block_content'] as $group_entity_type_id) {
+      for ($i = 0; $i < 2; $i++) {
```

Comment:
In the setup 2 node groups and 2 block_content groups were made. This iterates over all of them and checks if the results are correct.